[WEBSITE-400] Restructure the Jenkins User Documentation's 'User Handbook'.

gilesgas opened the pull request:
https://issues.jenkins-ci.org/browse/WEBSITE-400

This entails:

* Moving "Installing Jenkins" out from "Getting Started with Jenkins" and
into its own top-level chapter.
* Disabling the 'Best Practices' and 'Jenkins Use-Cases' chapters (until
more content can fill these empty chapters).
* Moving 'Managing Jenkins' and 'Jenkins System Administration' chapters
(formerly 'Operating Jenkins') further down into the handbook. Updating 4
links to the new 'system-administration' link.
* Moving 'Advanced System Administration' into the 'Installing Jenkins'
chapter near the top.
* Disabling the 'Fingerprints' and 'Remote API' topics (until more
content can fill these empty sections).

daniel-beck commented:
Perhaps this calls for redirects, or a custom 404 error page for /doc/handbook directing people to new chapter locations?

rtyler commented:
@gilesgas I think @daniel-beck's correct in that this change should include redirects from old locations to new locations. There are some files which have the `layout: redirect` (see `content/redirect/` for examples).

Basically creating a stub file where an old file existed, with the proper redirect layout means old permalinks to those locations will be correctly redirected, which is IMO important.

bitwiseman commented on content/doc/book/installing/installing-on-different-platforms.adoc:
```diff
@@ -8,7 +8,7 @@ layout: section
 :toc: left
 
-= Installing Jenkins
+= Installing Jenkins on Different Platforms
```
This is the first step.  Got it.

daniel-beck commented on content/doc/book/installing/installing-on-different-platforms.adoc:
```diff
@@ -8,7 +8,7 @@ layout: section
 :toc: left
 
-= Installing Jenkins
+= Installing Jenkins on Different Platforms
```
Why? I don't want to install Jenkins on different platforms. I just want to install it on one. Mine.

(Not actually trolling, I think, even though this might look like it.)

gilesgas replied:
Hi @daniel-beck and @bitwiseman ,

My rationale for taking @bitwiseman 's original "Installing Jenkins" page (as a _section_ page within the "Getting Started with Jenkins" _chapter_ at https://jenkins.io/doc/book/getting-started/installing/) and moving this page into its own _chapter_ called "Installing Jenkins" is documented in https://issues.jenkins-ci.org/browse/WEBSITE-400.

As someone who's still relatively new to Jenkins and its documentation (like me), it would be easier to find everything that relates to installing Jenkins in a single _chapter_ (rather than this content being fragmented across the handbook). Currently, everything documented about "Installing Jenkins" is nested within a "Getting Started with Jenkins" _chapter_ (which is fair enough), but then there's currently an empty "Advanced Jenkins Installation" _section_ page nested within the "Appendix" _chapter_. Assuming someone wants to add something to the "Advanced Jenkins Installation" page at some point, then for the reader, it would be easier to find everything that relates to installing Jenkins in a single _chapter_ - especially since the jenkins.io documentation currently has no page search feature.

My initial intention was to have a _chapter_ page called "Installing Jenkins" and within that _chapter_, have separate _section_ pages for "Installing Jenkins on Unix/Linux", "Installing Jenkins on Mac OS X", etc. This would entail breaking up the existing page's content into their appropriate pages. (The common bits of information between each platform - like the 'Pre-Install' portion on this page - could be placed in the top "Installing Jenkins" _chapter_ page.) However, I think all of this work (which I suspect would also cover https://issues.jenkins-ci.org/browse/WEBSITE-127) is outside the scope of this pull request.

Nevertheless, to start off this work, I initially decided to nest the existing "Installing Jenkins" _section_ page (currently within the "Getting Started with Jenkins" _chapter_) into a new "Installing Jenkins" _chapter_ and rename this _section_ page to "Installing Jenkins on Different Platforms". This was to avoid having an "Installing Jenkins" _chapter_ page with an "Installing Jenkins" _section_ page (i.e. the page with the existing content on it).

However, in retrospect, I now think this is overkill and is most likely confusing the matter. Therefore, I've now updated this pull request to effectively move the existing "Installing Jenkins" _section_ page into the "Installing Jenkins" _chapter_ page and have the empty "Advanced Jenkins Installation" page nested within this chapter.

At some point in the future, we can create separate "Installing Jenkins on Unix/Linux", "Installing Jenkins on Mac OS X", etc. pages within this new "Installing Jenkins" _chapter_ and migrate the content from this _chapter_ page into their appropriate _section_ pages.

I hope all this makes sense?

gilesgas replied:
Incidentally, I've placed in a redirect so that https://jenkins.io/doc/book/getting-started/installing/ will redirect to https://jenkins.io/doc/book/installing/ (i.e. the new _chapter_ page).

gilesgas commented:
Hi @daniel-beck and @rtyler ,

Thanks for the explanation Tyler - I've added in all the redirects I think are required, so that in theory, no existing link should break (but instead, now be directed to its correct URL). Incidentally, I have implemented these redirects as relative links (as opposed to full URLs). Hope this is okay?

@daniel-beck , afaik, we don't appear to have had a redirect from /doc/handbook to /doc/book before (if that's what you're implying above?) ... otherwise, I can add one if you feel this is necessary?

@rtyler - looking at the build logs, I think the reason that the builds of this pull request are failing is because I've [removed the "Best Practices" and "Jenkins Use-Cases" chapters from the User Handbook](https://github.com/jenkins-infra/jenkins.io/compare/master...gilesgas:WEBSITE-400#diff-ad75e9616f4f84e88d6b421aaa642d06) (which we discussed earlier in the [Google Groups discussion](https://groups.google.com/forum/#!topic/jenkinsci-docs/fI_gfQ0_ZWk)). However, I suspect the remaining presence of the "Best Practices" and "Jenkins Use-Cases" source content might be causing the build to fail? If so, do you have any suggestions on how I might rectify this situation?

gilesgas commented:
In retrospect - perhaps I should've broken up this work up into separate pull requests.

One thing I haven't been able to do is nest the "Pipelines" and "Blue Ocean" chapters within the "Using Jenkins" chapter (as documented in WEBSITE-400).

It looks like the User Handbook can only support 1 level of nesting (as opposed to 2, which is what the [FreeBSD Handbook](https://www.freebsd.org/doc/en_US.ISO8859-1/books/handbook/) has.

rtyler commented:
Had the opportunity to chat with @gilesgas via Google Hangouts about this pull request. I'm going to "adopt" this PR and slice and dice it where necessary for merging.

gilesgas commented:
Thank you @rtyler - I will aim to make my subsequent pull requests more digestible in future.

rtyler commented:
I've rebased and chopped this up a bit. Should be passing some tests now.

I made the Executive Decision :tm:   to simply deleted "Use Cases" and "Best Practices" without redirects since there wasn't content of any import in either "chapter" but rather stubs from @bitwiseman's and my initial brain dumping.
